packages/itwin/tree-widget/src/test/trees/models-tree/ModelsVisibilityHandler.test.ts

```diff
@@ -121,6 +123,37 @@ describe("ModelsVisibilityHandler", () => {
     });
   });
 
+  it("getNodeType", () => {
+    expect(ModelsVisibilityHandler.getNodeType({} as TreeNodeItem)).to.be.eq(ModelsTreeNodeType.Unknown);
+    const isClassGroupingNodeKeySpy = sinon.stub(NodeKey, "isClassGroupingNodeKey").returns(true);
+    const node = {
+      key: {
+        type: StandardNodeTypes.ECInstancesNode,
+        version: 0,
+        instanceKeys: [{ className: "MyDomain:SpatialCategory", id: "testInstanceId" }],
+        pathFromRoot: [],
+      },
+      id: "testId",
+      label: PropertyRecord.fromString("category-node"),
+      autoExpand: true,
+      hasChildren: true,
+    } as PresentationTreeNodeItem;
+    expect(ModelsVisibilityHandler.getNodeType(node)).to.be.eq(ModelsTreeNodeType.Grouping);
+    isClassGroupingNodeKeySpy.returns(false);
+    expect(ModelsVisibilityHandler.getNodeType(node)).to.be.eq(ModelsTreeNodeType.Unknown);
+    node.extendedData = {};
+    const isSubjectNodeSpy = sinon.stub(ModelsVisibilityHandler, "isSubjectNode").returns(true);
```

Comment:
I think it would be better to create nodes with different `extendedData` and keys instead of stubing `ModelsVisibilityHandler`